### PR TITLE
IPv6 only: add API option enable/disable IPv4

### PR DESCRIPTION
**Pull request by robmry**
**- What I did**

- Added top level network-create option EnableIPv4 (which can later be hooked up to an option `--ipv4` in the CLI, equivalent to `--ipv6`).
- Added equivalent driver option `com.docker.network.enable_ipv4`.
  - Like enable_ipv6, it can be set via `default-network-opts`.
- Show EnableIPv4 alongside EnableIPv6 in "inspect" output.
- Predefined Linux none/host and Windows null networks are created with `EnableIPv4=false`.
- Other networks, including predefined Linux bridge default to `EnableIPv4=true`.\
- Ignore (clear) `EnableIPv4` in a network create request if the API version is less than 1.47.
- Require `--experimental` to disable IPv4.

Follow-up PRs will make the option do-something.

**- How I did it**

The first commit here, bumping the API version to 1.47, is likely to disappear - as this will be merged after another PR that does the same thing. But, for now, it means there's somewhere to put an API `version-history.md` update.

The rest is fairly machanical copying of `EnableIPv6` behaviour.

**- How to verify it**

Can't disable IPv4 without `--experimental`:
```
# docker network create -o com.docker.network.enable_ipv4=false n1
Error response from daemon: IPv4 can only be disabled if experimental features are enabled
```

With `--experimental`, can disable IPv4:

```
# docker network create -o com.docker.network.enable_ipv4=false n1
e3fc95637aa2d9902ed188960283e025b95521cd44c95a645e0b48e7548ef2e6
# docker network inspect n1
[
    {
        "Name": "n1",
        ...
        "EnableIPv4": false,
        "EnableIPv6": false,
```

Default for a new bridge network is `true`:
```
# docker network create n2
9d71a90d0353fd352d53495fca214fa1a8ddecaca90c959a4ef9ed2318438d05
# docker network inspect n2
[
    {
        "Name": "n2",
        "Id": "9d71a90d0353fd352d53495fca214fa1a8ddecaca90c959a4ef9ed2318438d05",
        ...
        "EnableIPv4": true,
        "EnableIPv6": false,
```

Predefined host network shows `EnableIPv4:false`, like the existing `EnableIPv6:false`:
```
# docker network inspect host
[
    {
        "Name": "host",
        ...
        "EnableIPv4": false,
        "EnableIPv6": false,
```

Predefined bridge network has `EnableIPv4:true`:
```
# docker network inspect bridge
[
    {
        "Name": "bridge",
        ...
        "Driver": "bridge",
        "EnableIPv4": true,
        "EnableIPv6": false,
       ...
```

Marshalling/unmarshalling a `libnetwork.Network` with `EnableIPv4:true` is covered in an updated unit test.

**- Description for the changelog**
```markdown changelog
- API Changes:
  - `POST /networks/create` now has an `EnableIPv4` field. Setting it to `false`
  disables IPv4 IPAM for the network. It can only be set to `false` if the
  daemon has experimental features enabled.
  - `GET /networks/{id}` now returns an `EnableIPv4` field showing whether the
  network has IPv4 IPAM enabled.
```

### CODE REVIEW

**Review by corhere — APPROVED** [review]
LGTM! Small PRs for the win!

File: daemon/network.go [hunk 1]
```diff
@@ -316,18 +316,33 @@ func (daemon *Daemon) createNetwork(cfg *config.Config, create networktypes.Crea
...
+	enableIPv4 := create.ConfigFrom == nil
+	if create.EnableIPv4 != nil {
+		enableIPv4 = *create.EnableIPv4
+	} else if v, ok := networkOptions[netlabel.EnableIPv4]; ok {
+		var err error
+		if enableIPv4, err = strconv.ParseBool(v); err != nil {
+			return nil, errdefs.InvalidParameter(fmt.Errorf("driver-opt %q is not a valid bool", netlabel.EnableIPv4))
+		}
+	}
+	if !enableIPv4 && !daemon.config().Experimental && create.ConfigFrom == nil {
```
[the remark this answers is not in the document]
Comment (robmry) [reply daemon/network.go:328]: This PR on its own is basically-useless (!). It'll carry on being basically-useless for a few more PRs yet.

I don't think we've decided to release/announce IPv6-only (`EnableIPv4:false`) as experimental, or behind a feature flag - but, we could do either if we decide to release with it still incomplete. Depending on what that looks like, we might want to feature-flag it for a specific network driver or something like that. Or, if it looks risky and likely to change, we might make the whole thing experimental / feature-flagged until we get some feedback. But, I don't think we'll know until we're at that point.

So, just locking it behind `--experimental` for now seems like the best approach?

[the remark this answers is not in the document]
Comment (akerouanton) [reply daemon/network.go:328]: Fair enough, I won't dwell on that.

**Review by akerouanton — APPROVED** [review]
LGTM

**Review by corhere — APPROVED** [review]

**Review by akerouanton — APPROVED** [review]

**Pull request merged**